paritytech/parity-signer
# feat: Show accounts with unknown network

closes #389 

How to test:
- create a Kusama account (1)
- modify the genesisHash aka `networkKey` of Kusama in `src/constants.js`
- reload the App

Because the  genesisHash used to create (1) isn't known any more, we can't match the Kusama Network. Still this account can be seen, its backup retrieved with the pin and deleted.

edit:
This was quite a painful PR, trying to remove many weird things that were done. Now we use the same [key, value] mapping in the accountStore and in the db. This allows to show accounts which we have no corresponding `NetworkKey`. It also helps to seemlessly save or delete accounts (even if the network is not known) because the db and the store have the same keys.
 
This PR introduces the notion of `selectedKey` that corresponds to a `selectedAccount` (more broadly named `selected` in the app). I think it's quite straightforward, we should use this key as much as possible and only use `accountId()` when dealing with Data coming in (QR code scanning) or saving new accounts.

I tested the migration but please feel free to do it as well ([as explained here](https://github.com/paritytech/parity-signer/pull/352#issuecomment-527437669))

![image](https://user-images.githubusercontent.com/33178835/65086511-60072d80-d9b2-11e9-98d9-92c224e02006.png)
![image](https://user-images.githubusercontent.com/33178835/65086514-65fd0e80-d9b2-11e9-89fe-6e1327e0883c.png)
